src/golang/lib/workflow/operator/base.go

```diff
@@ -76,6 +78,11 @@ func unknownSystemFailureExecState(err error, logMsg string) *shared.ExecutionSt
 }
 
 func (bo *baseOperator) launch(ctx context.Context, spec job.Spec) error {
+	if bo.execState.Status != shared.PendingExecutionStatus {
+		return ErrInvalidStatusToLaunch
```

Comment:
One thing I find a bit annoying about these internal errors is that we generally don't templatize them (eg. add the operator name, invalid status, expected information to the error message), so the message isn't that helpful. I wonder if we should wrap our errors in a code instead, and compare that code value. That way, the error message can be as useful as possible. I guess what I'm getting at is that there isn't really a need to standardize internal errors right? As long as we can distinguish between different types of errors.This is of course something to do in the future, just curious about your opinion.

In the meantime, I wonder if we should just create the error directly with `errors.Newf()` and make the message as helpful as possible?

Reply:
I've been debating on this as well. I feel creating constant errors will be helpful to handle different types of errors in caller, and I'm imagining context could be given from logs.

For immediate need, I think your suggestion makes sense